backend/src/datasources/post.datasource.ts

```diff
@@ -0,0 +1,55 @@
+import * as uuid from 'uuid';
+import { DataSource } from 'apollo-datasource';
+import Post from '../entities/post.entity';
+import { posts } from '../seed-data'
+
+export default class PostDatasource extends DataSource {
+    private posts: Post[];
+
+    constructor() {
+        super();
+
+        this.posts = posts;
+    }
+
+    async getPosts() {
+        return Promise.resolve(this.posts);
+    }
+
+    async getPost(id: string) {
+        return Promise.resolve(this.posts.find(post => post.id === id));
+    }
+
+    async createPost(post: Partial<Post>) {
+        const fullPost : Post = {
+            id: uuid.v4(),
+            title: post.title,
+            voters: [], 
+            userName: post.userName
+        }
+
+        this.posts = [... this.posts, fullPost]
+
+        return Promise.resolve(fullPost);
+    }
+
+    async updatePost(id: string, post: Post) {
+        const index = this.posts.findIndex(post => post.id === id);
+        this.posts.splice(index, 1, post);
```

Comment:
Didn't know that you can also specify the item to add with [splice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice) I never used it because of in-place mutation. But now I think it actually not that bad :+1: